src/vdc_collection.rs

```diff
@@ -0,0 +1,272 @@
+        let raw = match self.storage.get(self.str_to_key(id)) {
+            Ok(x) => x.0,
+            Err(e) => return Err(VdcCollectionError::LoadFailed(e)),
+        };
+
+        match serde_cbor::de::from_slice(&raw) {
+            Ok(x) => Ok(x),
+            Err(_) => Err(VdcCollectionError::DeserializeFailed),
+        }
+    }
+
+    /// Remove a credential from the store.
+    pub fn delete(&self, id: &str) -> Result<(), VdcCollectionError> {
+        match self.storage.remove(self.str_to_key(id)) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(VdcCollectionError::DeleteFailed(e)),
+        }
+    }
+
+    /// Get a list of all the credentials.
+    pub fn all_entries(&self) -> Vec<Key> {
+        let mut r = Vec::new();
+
+        for key in self.storage.list() {
+            let name = key.0;
+
+            if name.starts_with(KEY_PREFIX) {
+                if let Some(id) = name.strip_prefix(KEY_PREFIX) {
+                    r.push(Key(id.to_string()));
+                }
+            }
+        }
+
+        r
+    }
+
+    /// Get a list of all the credentials that match a specified type.
+    pub fn all_entries_by_type(&self, ctype: CredentialType) -> Vec<String> {
+        let mut r = Vec::new();
+
+        for key in self.all_entries() {
+            let cred = self.get(&key.0);
+
+            if let Ok(x) = cred {
+                if x.ctype == ctype {
+                    r.push(key.0);
+                }
+            }
+        }
+
+        r
+    }
+
+    /// Convert a UUID to a storage key.
+    fn id_to_key(&self, id: Uuid) -> Key {
+        self.str_to_key(&id.to_string())
+    }
+
+    /// Convert a string ref to a storage key.
+    fn str_to_key(&self, id: &str) -> Key {
+        Key(format!("{}{}", KEY_PREFIX, id))
+    }
+
+    /// Dump the contents of the credential set to the logger.
+    pub fn dump(&self) {
+        let span = info_span!("All Credentials");
+        span.in_scope(|| {
+            for key in self.all_entries() {
+                if let Ok(x) = self.get(&key.0) {
+                    info!("{:?}", x);
+                }
+            }
+        });
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::local_store::*;
+    use uuid::uuid;
+
+    #[test]
+    fn test_vdc() {
+        let smi = LocalStore;
+        let vdc = VdcCollection::new(Box::new(smi));
+        let payload_1: Vec<u8> = "Some random collection of bytes. ⚛".into();
+        let payload_2: Vec<u8> = "Some other random collection of bytes. 📯".into();
+        let payload_3: Vec<u8> = "Some third random collection of bytes. λ".into();
+
+        match vdc.add(
+            uuid!("00000000-0000-0000-0000-000000000001"),
+            CredentialFormat::MsoMdoc,
+            CredentialType::Iso18013_5_1mDl,
+            payload_1.clone(),
+        ) {
+            Ok(_) => {}
+            Err(_) => panic!("Unable to add the first value."),
+        }
```

Comment:
We can use `.expect` on a `Result` type to core dump with a message.

```rust
vdc.add(
     uuid!("00000000-0000-0000-0000-000000000001"),
     CredentialFormat::MsoMdoc,
     CredentialType::Iso18013_5_1mDl,
     payload_1.clone(),
).expect("Unable to add the first value.");
```

Reply:
Fixed.